### PR TITLE
Add typescript definition file

### PR DESCRIPTION
**Pull request by koddsson**
Ref: github/web-systems#223

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export default class RemoteInputElement extends HTMLElement {
+  readonly input?: HTMLInputElement;
+  src: string;
+}
```
Comment (muan) [inline index.d.ts]: EOF.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default class RemoteInputElement extends HTMLElement {
+  readonly input?: HTMLInputElement;
```
Comment (muan) [inline index.d.ts]: What does `input?` do here? We seem to do this elsewhere:

```suggestion
  readonly input: HTMLInputElement | undefined;
```

And the flow def is actually incorrect in that we should accept `HTMLTextAreaElement` too:

https://github.com/github/remote-input-element/blob/2b899d0044671e9f17e1ec45569dd1cf152584d2/index.js#L41-L44

Comment (koddsson) [reply index.d.ts]: Sorry this was due to me having a really old VSCode that had outdated TypeScript installed.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export default class RemoteInputElement extends HTMLElement {
+  readonly input: HTMLInputElement | HTMLTextAreaElement | undefined;
+  src: string;
```
Comment (muan) [inline index.d.ts]: The setter is not typed.

Comment (muan) [reply index.d.ts]: nvm https://github.com/Microsoft/TypeScript/issues/3745.

**Pull request merged**